graylog2-server/src/main/java/org/graylog2/database/dbcatalog/DbEntitiesScanner.java

```diff
@@ -52,9 +56,48 @@ public DbEntitiesScanner(final ChainingClassLoader chainingClassLoader) {
     @Override
     public DbEntitiesCatalog get() {
         final Stopwatch stopwatch = Stopwatch.createStarted();
-        final DbEntitiesCatalog catalog = dbEntitiesScanningMethod.scan(packagesToScan, packagesToExclude, chainingClassLoader);
+        final DbEntitiesCatalog catalog = scan(packagesToScan, packagesToExclude, chainingClassLoader);
         stopwatch.stop();
         LOG.info("{} entities have been scanned and added to DB Entity Catalog, it took {}", catalog.size(), stopwatch);
         return catalog;
     }
+
+    public DbEntitiesCatalog scan(final String[] packagesToScan,
+                                  final String[] packagesToExclude,
+                                  final ChainingClassLoader chainingClassLoader) {
+        List<DbEntityCatalogEntry> dbEntities = new LinkedList<>();
+        ClassGraph classGraph = new ClassGraph()
+                .enableAnnotationInfo()
+                .acceptPackages(packagesToScan)
+                .rejectPackages(packagesToExclude)
+                .filterClasspathElements(classpathElementPathStr -> classpathElementPathStr.contains("graylog"))
+                .disableRuntimeInvisibleAnnotations();
+
+        if (chainingClassLoader != null) {
+            //Unfortunately, ClassGraph does not work correctly if provided with ChainingClassLoader as a whole
+            //You have to manually add all class loaders from ChainingClassLoader
+            for (ClassLoader cl : chainingClassLoader.getClassLoaders()) {
+                classGraph = classGraph.addClassLoader(cl);
+            }
+        }
+
+        try (ScanResult scanResult = classGraph.scan()) {
+            final String annotationName = DbEntity.class.getCanonicalName();
+            ClassInfoList classInfoList = scanResult.getClassesWithAnnotation(annotationName);
+            for (ClassInfo classInfo : classInfoList) {
+                final var annotations = classInfo.getAnnotationInfoRepeatable(annotationName);
+                for (final var annotationInfo : annotations) {
+                    AnnotationParameterValueList paramVals = annotationInfo.getParameterValues();
+                    dbEntities.add(new DbEntityCatalogEntry(
+                            paramVals.get("collection").getValue().toString(),
+                            paramVals.get("titleField").getValue().toString(),
+                            classInfo.loadClass(),
+                            paramVals.get("readPermission").getValue().toString()
+                    ));
+                }
+            }
+
+            return new DbEntitiesCatalog(dbEntities);
```

Comment:
It looks like `DbEntitiesCatalog` isn't prepared to handle multiple entries for the same class because it keeps them in a map: 

https://github.com/Graylog2/graylog2-server/blob/4d46d6e63f68097ab8c6c8ac948914c6a9e97446/graylog2-server/src/main/java/org/graylog2/database/dbcatalog/DbEntitiesCatalog.java#L48

We should fix that or remove the `getByModelClass` method, as it appears unused.

Reply:
Removing it in #19800.